### PR TITLE
add support for watching entities

### PR DESCRIPTION
**Pull request by simon3z**

### CODE REVIEW

File: test/kubeclient_test.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 require 'webmock/minitest'
 require './lib/kubeclient'
 
+TEST_FILE_PATH = File.dirname(__FILE__)
```
Comment (abonas) [inline test/kubeclient_test.rb]: why is this needed? how are you invoking tests?

Comment (simon3z) [inline test/kubeclient_test.rb]: I am invoking it with:

    rake test

but obviously I don't want to embed fixed paths such as:

    json_watch_stream = File.new('test/watch_stream.json')

I prefer to discover the path and use:

    json_watch_stream = File.new(File.join(TEST_FILE_PATH, 'watch_stream.json'))

so it's possible to invoke rake from any path and there's no assumption on where the code, Rakefile and tests are.

File: lib/kubeclient.rb [hunk 2]
```diff
@@ -8,9 +9,36 @@
 require 'kubeclient/replication_controller'
 require 'kubeclient/entity_list'
 require 'kubeclient/kube_exception'
-
+require 'kubeclient/watch'
 
 module Kubeclient
+
+  # HTTP Stream used to watch changes on entities
+  class WatchStream
```
Comment (abonas) [inline lib/kubeclient.rb]: afaik it's considered best practice to put one class per file - that is to put WatchStream in a separate file.
But I'd like to hear other opinions as well. @tenderlove ? @Fryguy ?

**Comment by abonas** [comment]
can you also update the readme with usage example for watch?

File: lib/kubeclient.rb (hunk 2 above)
Comment (simon3z) [inline lib/kubeclient.rb]: Many modules in the stdlib contain multiple classes per file. In rails instead it's discouraged/prohibited because it's breaking its own autoloading mechanism.

In our case I think it boils down to whether it belongs there or not logically: in my opinion it's part of the rest-api client engine that currently reside in that file.

Comment (Fryguy) [inline lib/kubeclient.rb]: Yes, it is very common practice for one class per file, generally in a subdirectory structure that matches the underscorized full namespace.  Then, usually a top file requires it.

So, Kubeclient::WatchStream generally lives in lib/kubeclient/watch_stream.rb and in lib/kubeclient.rb you would `require 'kubeclient/watch_stream'`
